### PR TITLE
refactor SmartAction and MoveRunner

### PR DESCRIPTION
**Pull request by huafengw**
Please **DO NOT** merge this pr because it's not ready

### CODE REVIEW

**Comment by huafengw** [comment]
postpone merging this pr

File: smart-actions/src/main/java/org/smartdata/actions/hdfs/CheckStorageAction.java [hunk 1]
```diff
@@ -43,13 +43,11 @@ public void init(Map<String, String> args) {
...
   @Override
-  protected void execute() {
-    ActionStatus actionStatus = getActionStatus();
-    actionStatus.begin();
+  protected void execute() throws ActionException {
     try {
       HdfsFileStatus fileStatus = dfsClient.getFileInfo(fileName);
       if (fileStatus == null) {
-        throw new IllegalArgumentException("File does not exit.");
+        throw new ActionException("File does not exit.");
```
Comment (drankye) [inline smart-actions/src/main/java/org/smartdata/actions/hdfs/CheckStorageAction.java]: exit -> exist

File: smart-actions/src/main/java/org/smartdata/actions/hdfs/CheckStorageAction.java [hunk 2]
```diff
@@ -72,12 +70,9 @@ protected void execute() {
         blockInfo.append("}");
         this.resultOut.println(blockInfo);
       }
-      actionStatus.setSuccessful(true);
     } catch (Exception e) {
-      actionStatus.setSuccessful(false);
       LOG.error("CheckStorageAction failed", e);
-    } finally {
-      actionStatus.end();
+      throw new ActionException(e);
```
Comment (drankye) [inline smart-actions/src/main/java/org/smartdata/actions/hdfs/CheckStorageAction.java]: Hmm, maybe bad to throw exception in finally?

File: smart-actions/src/main/java/org/smartdata/actions/hdfs/MoveFileAction.java [hunk 3]
```diff
@@ -58,27 +59,24 @@ public void init(Map<String, String> args) {
...
     logOut.println("Action starts at "
         + (new Date(System.currentTimeMillis())).toString() + " : "
         + fileName + " -> " + storagePolicy.toString());
     try {
       dfsClient.setStoragePolicy(fileName, storagePolicy);
     } catch (Exception e) {
-      actionStatus.end();
-      actionStatus.setSuccessful(false);
-      throw new RuntimeException(e);
+      throw new ActionException(e);
```
Comment (drankye) [inline smart-actions/src/main/java/org/smartdata/actions/hdfs/MoveFileAction.java]: How about:
throw new ActionException("Failed to setStoragePolicy for file {} with storage policy {}", e);

File: smart-actions/src/main/java/org/smartdata/actions/hdfs/MoveFileAction.java [hunk 4]
```diff
@@ -58,27 +59,24 @@ public void init(Map<String, String> args) {
...
-    moveRunner = new MoverBasedMoveRunner(
-        getContext().getConf(), getActionStatus());
+    moveRunner = new MoverBasedMoveRunner(getContext().getConf(), getActionStatus());
 
     try {
       moveRunner.move(fileName);
-      actionStatus.setSuccessful(true);
     } catch (Exception e) {
-      throw new RuntimeException(e);
+      throw new ActionException(e);
```
Comment (drankye) [inline smart-actions/src/main/java/org/smartdata/actions/hdfs/MoveFileAction.java]: throw new ActionException("Calling moveRunner.move() failed for fileName={}", e);

File: smart-actions/src/main/java/org/smartdata/actions/hdfs/move/MoverBasedMoveRunner.java [hunk 5]
```diff
@@ -41,15 +41,17 @@ public MoverBasedMoveRunner(Configuration conf, ActionStatus actionStatus) {
   }
 
   @Override
-  public void move(String file) throws IOException {
+  public void move(String file) throws IOException, InterruptedException {
```
Comment (drankye) [inline smart-actions/src/main/java/org/smartdata/actions/hdfs/move/MoverBasedMoveRunner.java]: Don't quite like throwing the InterruptedException

File: smart-actions/src/main/java/org/smartdata/actions/hdfs/CheckStorageAction.java (hunk 2 above)
Comment (huafengw) [inline smart-actions/src/main/java/org/smartdata/actions/hdfs/CheckStorageAction.java]: It's not in the finally block.

**Comment by drankye** [comment]
Let's have a code review discussion tmr for this, since the change is very big.

**Pull request merged**